chess/src/game.rs

```diff
@@ -140,7 +141,7 @@ impl Game {
         if self.check_for_draw() {
             return;
         }
-        let king_is_under_attack = self.is_king_under_attack();
+        let king_is_under_attack = self.history.get_real_turn().king_is_checked;
```

Comment:
You sure this works? I think we need to change the player. Probably can still call `is_king_under_attack`.

Reply:
Why would we want to change the player? `king_is_checked` is just a bool. We change the player when we actually check to see if the king is in check.. in `is_king_under_attack`